File: day_trader/sources/strategy/strategies/day_trader/strategy/services/websocket/common/DTO/subscriptions/websocket_subscribe_message.py
from sources.strategy.strategies.day_trader.strategy.services.websocket.common.DTO.websocket_message import *


class WebSocketSubscribeMessage(WebSocketMessage):


    @staticmethod
    def _SUBSCRIPTION_TYPE_UNSUBSCRIBE():
        return "U"

    @staticmethod
    def _SUBSCRIPTION_TYPE_SUBSCRIBE():
        return "S"

    def __init__(self, Msg ,SubscriptionType ,Service,ServiceKey,UUID=None, ReqId = None):
        super(WebSocketSubscribeMessage, self).__init__(Msg)
        self.UUID = UUID
        self.SubscriptionType = SubscriptionType
        self.Service = Service
        self.ServiceKey = ServiceKey
        self.ReqId = ReqId

File: test_clients/C#/day_trader_testclient/DayTraderTestClient/DayTraderTestClient.Common/DTO/Subscription/WebSocketSubscribeMessage.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DayTraderTestClient.Common.DTO.Subscription
{
    public class WebSocketSubscribeMessage:WebSocketMessage
    {
        #region Public Static Consts

        public static string _SUSBSCRIPTION_TYPE_SUBSCRIBE = "S";

        public static string _SUSBSCRIPTION_TYPE_UNSUBSCRIBE = "U";

        #endregion

        #region Public Attributes

        public string SubscriptionType { get; set; }

        public string UUID { get; set; }

        public string Service { get; set; }

        public string ServiceKey { get; set; }

        public string ReqId { get; set; }

        #endregion
    }
}
